.github/workflows/check-source-signatures.yml

```diff
@@ -0,0 +1,111 @@
+
+      - name: Prepare the build environment
+        run: |
+          if [ -z "${{ env.changed_pkgs }}" ] && [ -z "${{ env.changed_pkgs_extended }}" ]; then
+            echo "No package changes detected."
+            exit 0
+          fi
+
+          echo "Checking for invalid signatures..."
+          # Call this script to sync the toolchain manifests with the LKG daily build.
+          ./toolkit/scripts/setuplkgtoolchain.sh
+          # Determine the LKG daily build ID.
+          LKG_BUILD_ID=$(wget -qO - https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json | jq -r ".dailybuildid" | tr '\.' '-')
+          echo "LKG_BUILD_ID=${LKG_BUILD_ID}" >> $GITHUB_ENV
+
+          sudo make -C toolkit -j$(nproc) chroot-tools REBUILD_TOOLS=y DAILY_BUILD_ID=${LKG_BUILD_ID}
+
+      - name: Check for invalid source signatures
+        run: |
+          if [ -z "${{ env.changed_pkgs }}" ] && [ -z "${{ env.changed_pkgs_extended }}" ]; then
+            echo "No package changes detected."
+            exit 0
+          fi
+
+          # Core SPECs
+          if [ -n "${{ env.changed_pkgs }}" ]; then
+            # We want to ignore errors here, as we want to check all the packages that have been modified. Capture the error code and check it later.
+            set +e
+            echo "Running: sudo make -C toolkit input-srpms REBUILD_TOOLS=y DAILY_BUILD_ID=${{ env.LKG_BUILD_ID }} SRPM_PACK_LIST='${{ env.changed_pkgs }}'"
+            sudo make -C toolkit -j$(nproc) input-srpms REBUILD_TOOLS=y DAILY_BUILD_ID=${{ env.LKG_BUILD_ID }} SRPM_PACK_LIST="${{ env.changed_pkgs }}"
+            core_err=$?
+            set -e
+          fi
+
+          # Extended SPECs
+          if [ -n "${{ env.changed_pkgs_extended }}" ]; then
+            # We want to ignore errors here, as we want to check all the packages that have been modified. Capture the error code and check it later.
+            set +e
+            echo "Running: sudo make -C toolkit input-srpms REBUILD_TOOLS=y DAILY_BUILD_ID=${{ env.LKG_BUILD_ID }} SRPM_PACK_LIST='${{ env.changed_pkgs_extended }}' SPECS_DIR=../SPECS-EXTENDED"
+            sudo make -C toolkit -j$(nproc) input-srpms REBUILD_TOOLS=y DAILY_BUILD_ID=${{ env.LKG_BUILD_ID }} SRPM_PACK_LIST="${{ env.changed_pkgs_extended }}" SPECS_DIR=../SPECS-EXTENDED
+            extended_err=$?
+            set -e
+          fi
+
+          # Print results
+          if [ $core_err -ne 0 ] || [ $extended_err -ne 0 ]; then
+            printf "\n\n******************************"
+            echo "Failed to check the signatures of the modified packages."
+            echo "Check the logs above for details on the mismatches files and their expected hashes."
+            if [ $core_err -ne 0 ]; then
+              echo "Consider running: sudo make -C toolkit input-srpms REBUILD_TOOLS=y SRPM_PACK_LIST='${{ env.changed_pkgs }}'"
+            elif [ $extended_err -ne 0 ]; then
```

Comment:
this should be a separate if statement, in case both SPECS and SPECS-EXTENDED have issues

Reply:
good point.